src/coreclr/src/tools/Common/TypeSystem/Common/MetadataFieldLayoutAlgorithm.cs

```diff
@@ -517,11 +517,15 @@ protected ComputedInstanceFieldLayout ComputeAutoFieldLayout(MetadataType type,
                 else
                 {
                     var fieldSizeAndAlignment = ComputeFieldSizeAndAlignment(fieldType, packingSize);
+                    largestAlignmentRequired = LayoutInt.Max(fieldSizeAndAlignment.Alignment, largestAlignmentRequired);
```

Comment:
This needs to go above the cascading `if` block, together with the line above it.